src/utilsplitstring.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <iterator>
+#include <set>
+#include <string>
+#include <vector>
+#include <type_traits>
+
+#ifndef BITCOIN_UTILSPLITSTRING_H
+#define BITCOIN_UTILSPLITSTRING_H
+
+/**
+ * Tokenizes a string by any of the given separators.
+ * @param[in] tokens            The container (either an instance of std::vector<std::string>
+ *                              or std::set<std::string>) to add tokenized string parts to.
+ * @param[in] str               The string to tokenize.
+ * @param[in] anyOfSeparator    A string with valid separators.
+ * @param[in] mergeEmpty        Set to true to merge adjacent separators (empty tokens); otherwise false (default).
+ */
+template<typename ContainerT>
+void Split(ContainerT& tokens, const std::string& str, const std::string& anyOfSeparator, bool mergeEmpty = false)
+{
+    static_assert(
+        std::is_same<std::vector<std::string>, ContainerT>::value ||
+        std::is_same<std::set<std::string>, ContainerT>::value,
+        "ContainerT must be of type std::vector<std::string> or std::set<std::string>");
+
+    auto insertIt = std::inserter(tokens, tokens.end());
+    if (str.empty()) {
+        *insertIt = "";
+        return;
+    }
+
+    const auto begin = str.cbegin();
+    const auto end = str.cend();
+
+    for (auto it = begin; it < end;) {
+        bool foundSeparator = false;
+        auto tokenIt = it;
```

Comment:
nit: this would be a bit simpler in implementation if it always referred to the end of the token, and you stashed the beginning here and continued to operate on it. Does that make sense?

```C++
for (auto it = begin; it < end; ++it) {
        bool foundSeparator = false;
        auto start = it;
        while (it < end && !(foundSeparator = anyOfSeparator.find(*it) != std::string::npos)) {
            ++it;
        }
        if (it != begin && (!mergeEmpty || start != it)) {
            *insertIt = std::string(start, it);
        }
        if (foundSeparator && (it == begin || it == end - 1)) {
            *insertIt = "";
        }
    }
```

Reply:
Thanks for the review @Empact!

I have updated the PR and addressed your feedback.

Please note that the separate `if` statements below are intentional and cover the case where `it` equals `begin` and `it` equals `end - 1`, in which case two empty tokens need to be inserted to be boost compliant.

```c++
        if (it == begin) {
            *insertIt = "";
        }
        if (it == end - 1) {
            *insertIt = "";
        }
```
This behavior is validated by the following unit test in `utilsplitstring_tests.cpp`:
```c++
       Split(result, ",", ",", true);
       std::vector<std::string> expected = {"", ""};
```